ui/src/components/Map/MapAlarmsGrid.vue

```diff
@@ -0,0 +1,343 @@
+  gridApi = params.api
+  gridColumnApi = params.columnApi;
+  autoSizeAll(false);
+}
+
+function autoSizeAll(skipHeader: boolean) {
+  var allColumnIds: string[] = [];
+  gridColumnApi.getAllColumns().forEach(function (column: any) {
+    allColumnIds.push(column.colId);
+  });
+  gridColumnApi.autoSizeColumns(allColumnIds, skipHeader);
+}
+
+watch(
+  () => [interestedNodesID.value, alarms.value],
+  () => {
+    gridApi.setRowData(
+      getAlarmsFromSelectedNodes()
+    );
+  }
+)
+
+function getAlarmsFromSelectedNodes() {
+  let alarms = store.getters['mapModule/getAlarmsFromSelectedNodes'];
+  return alarms.map((alarm: Alarm) => ({
+    id: +alarm.id,
+    severity: alarm.severity,
+    node: alarm.nodeLabel,
+    uei: alarm.uei,
+    count: +alarm.count,
+    lastEventTime: alarm.lastEvent.time,
+    logMessage: alarm.logMessage,
+  }));
+}
+
+const alarmOptions = ref(["Not Selected", "Acknowledge", "Unacknowledge", "Escalate", "Clear"]);
+
+let alarmOption = ref(alarmOptions.value[0]);
+
+watch(
+  () => [alarmOption.value],
+  (newValue, oldValue) => {
+    console.log("alarmOption changed from " + oldValue + " to " + newValue);
+    let alarmQueryParameters: AlarmQueryParameters;
+    switch (alarmOption.value) {
+      case alarmOptions.value[0]:
+        break;
+      case alarmOptions.value[1]: { // "Acknowledge"
+        alarmQueryParameters = { ack: true };
+        break;
+      }
+      case alarmOptions.value[2]: { // "Unacknowledge"
+        alarmQueryParameters = { ack: false };
+        break;
+      }
+      case alarmOptions.value[3]: { // "Escalate"
+        alarmQueryParameters = { escalate: true };
+        break;
+      }
+      case alarmOptions.value[4]: { // "Clear"
+        alarmQueryParameters = { clear: true };
+        break;
+      }
+      default:
+        console.log("No such alarm option exists: " + alarmOption.value);
+        break;
+    }
+
+    let numFail = 0;
+    let respCollection: any = [];
+    selectedAlarmIds.forEach((alarmId: string) => {
+      let resp = store.dispatch("mapModule/modifyAlarm", {
+        pathVariable: alarmId, queryParameters: alarmQueryParameters
+      })
+      respCollection.push(resp)
+    })
+    Promise.all(respCollection).then(function (result) {
+      result.forEach(r => {
+        if (r === false) {
+          numFail = numFail + 1;
+        }
+      })
+      GStore.flashMessage = (selectedAlarmIds.length - numFail) + " success, " + numFail + ' failed.'
+
+      setTimeout(() => {
+        GStore.flashMessage = ''
+        window.location.reload()
+      }, 4000)
+    })
+  }
+)
+
+const GStore: any = inject('GStore');
+
+let selectedAlarmIds: any = ref([]);
+
+let hasAlarmSelected: any = ref(false);
+
+function onSelectionChanged() {
+  let selectedRows = gridApi.getSelectedNodes().map((node: any) => node.data);
+  selectedAlarmIds = selectedRows.map((alarm: any) => alarm.id);
+  hasAlarmSelected.value = selectedAlarmIds.length > 0;
```

Comment:
`selectedAlarmIds` is a ref so should this be using `selectedAlarmIds.value` on 175 and 176?

One way to makes sure TS catches this is to avoid the use of `any` wherever possible. 
Above we have `let selectedAlarmIds: any = ref([]);` If we want to give this ref type safety we can change it like so `const selectedAlarmIds = ref<string[]>([]); `.

Reply:
Good point, updated.